lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -925,7 +1069,9 @@ void PatchEntryPointMutate::addSpecialUserDataArgs(SmallVectorImpl<UserDataArg>
   } else if (m_shaderStage == ShaderStageCompute) {
     // Emulate gl_NumWorkGroups via user data registers.
     // Later code needs to ensure that this starts on an even numbered register.
-    if (builtInUsage.cs.numWorkgroups || userDataUsage->isSpecialUserDataUsed(UserDataMapping::Workgroup)) {
+    // Always enable this, even if unused, if compute library is in use.
+    if (isComputeWithCalls() || builtInUsage.cs.numWorkgroups ||
+        userDataUsage->isSpecialUserDataUsed(UserDataMapping::Workgroup)) {
```

Comment:
Is `addSpecialUserDataArgs` called for computeWithCalls? It looks like it is only called for `isGraphics` shaders.

Reply:
I think it is called (via processComputeFuncs) for a compute pipeline or library.